model/jpa/src/main/java/org/keycloak/models/jpa/JpaClientProviderFactory.java

```diff
@@ -59,7 +64,20 @@ public void init(Config.Scope config) {
 
     @Override
     public void postInit(KeycloakSessionFactory factory) {
+        if (Profile.isFeatureEnabled(Profile.Feature.ADMIN_FINE_GRAINED_AUTHZ_V2)) {
```

Comment:
It would be better if we handle the event and provision the client within the scope of some authorization-specific SPI. We should review this once we have more discussions about such SPI and how it will play out with FGAP.